### PR TITLE
Bug fix in dynamic quantization kernels + better test coverage.

### PR DESCRIPTION
**Pull request by AshkanAliabadi**

### CODE REVIEW

File: aten/src/ATen/native/quantized/cpu/qnnpack/src/q8gemm/8x8-dq-aarch64-neon.S [hunk 1]
```diff
@@ -45,6 +45,11 @@ BEGIN_FUNCTION pytorch_q8gemm_dq_ukernel_8x8__aarch64_neon
     # Load multiplier
     LD1R {v26.4s}, [x8]
 
+    # v8 := zero
+    EOR v8.16b, v8.16b, v8.16b
```
Comment (kimishpatel) [inline aten/src/ATen/native/quantized/cpu/qnnpack/src/q8gemm/8x8-dq-aarch64-neon.S]: Not clear what was the bug here. I see v8 and v9 getting overwritten elsewhere, so I am not sure of the need to zero these guys.

**Review by kimishpatel — COMMENTED** [review]
Asked for clarifying comment.

File: aten/src/ATen/native/quantized/cpu/qnnpack/src/q8gemm/8x8-dq-aarch64-neon.S (hunk 1 above)
Comment (AshkanAliabadi) [reply aten/src/ATen/native/quantized/cpu/qnnpack/src/q8gemm/8x8-dq-aarch64-neon.S]: SMLAL is a (signed long) multiply-accumulate instruction.  v8 and v9 are used as accumulators there and should have been initialized to zero.

**Review by supriyar — COMMENTED** [review]
QNNPACK tests seem to be failing internally with this PR. Please address those as well. Thanks

**Comment by AshkanAliabadi** [comment]
Two tests were failing even before this PR and those breakages are not caused by this PR as far as I can tell.  My plate is full at this point so if you want them fixed you would have to wait unfortunately.

**Comment by supriyar** [comment]
> Two tests were failing even before this PR and those breakages are not caused by this PR as far as I can tell. My plate is full at this point so if you want them fixed you would have to wait unfortunately.
> 
> Edit: To be clear, those failures were there before. It's just that they were not exercised in the unit test and they are showing up now that I am adding extra tests. I can either disable those tests and come back to it later, or we can wait until I get around to fixing them. Sorry, it's just that I have a ton to do right now and I am running low on bandwidth.

I would rather that it not block this PR. Let's disable the failing tests for now and file a github issue/task so that it can be looked at later and we don't forget about it.

**Review by supriyar — APPROVED** [review]

**Pull request closed without merging**